### PR TITLE
[FIX 27#1] Add gh-pages corresponding to gnupg-2.2 branch

### LINKED ISSUE #27: Test and fix the branch gnupg-2.2
It should be adapted to work well with GnuPG-2.2

It should be useful to study the latest changes on GnuPG-2.2 (compared to GnuPG-2.1), and the latest version of the GnuPG docs, in order to check that maybe some of the things that EasyGnuPG tries to simplify are already simplified by GnuPG itself. This would make the code of EasyGnuPG lighter, less tricky, etc.

### PR DESCRIPTION
**Pull request by diveshuttam**
fixes partly  #27

### CODE REVIEW

File: _includes/readme-gnupg-2.2.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Installation
+
+    git clone --branch gnupg-2.2 https://github.com/easygnupg/egpg
+    cd egpg/
+    sudo make install
+
+## Requirements
+
+ - Ubuntu 16.04:
```
Comment (dashohoxha) [inline _includes/readme-gnupg-2.2.md]: Change this to Ubuntu 18.04

File: _includes/readme-gnupg-2.2.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        apt install gnupg2 pinentry-tty haveged libgfshare-bin parcimonie \
+               qrencode imagemagick zbar-tools wget realpath psmisc
+
```
Comment (dashohoxha) [inline _includes/readme-gnupg-2.2.md]: Replace realpath with coreutils.

In general, this file is an exact copy of README.md of the branch gnupg-2.2

File: gnupg-2.2/index.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+layout: readme-2.2
+---
+
+{% capture readme %}{% include readme-gnupg-2.1.md %}{% endcapture %}
```
Comment (dashohoxha) [inline gnupg-2.2/index.html]: Replace 2.1 with 2.2 here.

**Comment by diveshuttam** [comment]
@dashohoxha I have copied the readme from the gnupg-2.2 branch instead and added the versions as well

**Comment by dashohoxha** [comment]
Everything seems OK to me.
If you have already tested it with Jekyll I can accept it. Otherwise let us wait until I add jekyll support to egpg-ds container, so that you can test it.

**Comment by dashohoxha** [comment]
Now you should be able to test the gh-pages locally.
Try `ds make`, then `ds jekyll`, then open in browser `http://0.0.0.0:4000`.

**Comment by dashohoxha** [comment]
You also have to add the line `PORTS="4000:4000"` to `settings.sh`, before `ds make`.

**Comment by diveshuttam** [comment]
EDIT: I was able to resolve the following by setting up the container from the beginning. Though still working on connecting to the website outside the container. 
@dashohoxha I am getting seg fault while running `ds make` after pulling jekyll update. Not sure what they are about. Do you have some Idea?? (The packages don't show download much because this is the second time I ran this and were cached by docker)
```
Sending build context to Docker daemon  1.005MB
Step 1/7 : FROM ubuntu:18.04
 ---> 452a96d81c30
Step 2/7 : RUN apt update &&     apt -y upgrade &&     apt -y install systemd &&     systemctl set-default multi-user.target
 ---> Using cache
 ---> 1e423c12e137
Step 3/7 : CMD ["/sbin/init"]
 ---> Using cache
 ---> a9967b54b190
Step 4/7 : WORKDIR /host/egpg
 ---> Using cache
 ---> c6d4e841cf48
Step 5/7 : RUN apt -y install gnupg2 pinentry-tty haveged libgfshare-bin parcimonie         qrencode imagemagick zbar-tools wget coreutils psmisc make
 ---> Using cache
 ---> a72fc03db950
Step 6/7 : RUN apt -y install ruby-dev gcc &&     gem install ronn
 ---> Using cache
 ---> 2f83086bc6e8
Step 7/7 : RUN apt -y install jekyll
 ---> Using cache
 ---> 4e80ac25e1ac
Successfully built 4e80ac25e1ac
Successfully tagged egpg-bionic:latest

/usr/bin/ds: line 35: 13568 Segmentation fault      (core dumped) ds create

--> Running script '/usr/lib/ds/scripts/set_prompt.sh'
env: '/host/tmp/set_prompt.sh': No such file or directory

egpg-bionic
```

EDIT: the debug output
```
+ main make
+ set -o pipefail
+ VERSION=1.0
+ LIBDIR=/usr/lib/ds
+ PROGRAM=ds
+ source /usr/lib/ds/auxiliary.sh
+ [[ make != \r\u\n ]]
++ docker --version
++ cut -d, -f1
++ cut '-d ' -f3
++ cut -d. -f1
+ local version=18
+ [[ 18 -lt 17 ]]
+ [[ make == \c\d ]]
+ load_ds_config
+ DSDIR=/home/divesh/.ds
+ local config_file=/home/divesh/.ds/config.sh
+ [[ ! -f /home/divesh/.ds/config.sh ]]
+ unset GITHUB APPS CONTAINERS NETWORK
+ source /home/divesh/.ds/config.sh
++ GITHUB=https://github.com/docker-scripts
++ APPS=/opt/docker-scripts
++ CONTAINERS=/var/ds
++ NETWORK=dsnet
+ GITHUB=https://github.com/docker-scripts
+ APPS=/opt/docker-scripts
+ CONTAINERS=/var/ds
+ NETWORK=dsnet
+ mkdir -p /opt/docker-scripts /var/ds
+ local arg1=make
+ shift
+ case $arg1 in
+ load_container_settings
+ [[ -f settings.sh ]]
+ source ./settings.sh
++ APP=egpg-bionic
++ IMAGE=egpg-bionic
++ CONTAINER=egpg-bionic
++ PORTS=4000:4000
+ [[ -n egpg-bionic ]]
+ APP_DIR=/opt/docker-scripts/egpg-bionic
+ [[ -d /opt/docker-scripts/egpg-bionic ]]
+ [[ -d /opt/docker-scripts/egpg-bionic ]]
+ [[ -n egpg-bionic ]]
+ [[ -n egpg-bionic ]]
+ [[ -f /home/divesh/.ds/ds.sh ]]
+ [[ -f /opt/docker-scripts/egpg-bionic/ds.sh ]]
+ [[ -f ds.sh ]]
+ local command=make
+ case $command in
+ cmd_make
+ ds build
Sending build context to Docker daemon  1.012MB
Step 1/7 : FROM ubuntu:18.04
 ---> 452a96d81c30
Step 2/7 : RUN apt update &&     apt -y upgrade &&     apt -y install systemd &&     systemctl set-default multi-user.target
 ---> Using cache
 ---> 1e423c12e137
Step 3/7 : CMD ["/sbin/init"]
 ---> Using cache
 ---> a9967b54b190
Step 4/7 : WORKDIR /host/egpg
 ---> Using cache
 ---> c6d4e841cf48
Step 5/7 : RUN apt -y install gnupg2 pinentry-tty haveged libgfshare-bin parcimonie         qrencode imagemagick zbar-tools wget coreutils psmisc make
 ---> Using cache
 ---> a72fc03db950
Step 6/7 : RUN apt -y install ruby-dev gcc &&     gem install ronn
 ---> Using cache
 ---> 2f83086bc6e8
Step 7/7 : RUN apt -y install jekyll
 ---> Using cache
 ---> 4e80ac25e1ac
Successfully built 4e80ac25e1ac
Successfully tagged egpg-bionic:latest
+ ds create
/usr/bin/ds: line 35: 18282 Segmentation fault      (core dumped) ds create
+ ds config

--> Running script '/usr/lib/ds/scripts/set_prompt.sh'
env: '/host/tmp/set_prompt.sh': No such file or directory
egpg-bionic
```

EDIT 2:
On doing `strace` on `ds create` I just get a lot of brk() calls. i.e `ds create` just demands a lot of memory in just a short time and then a SIGSEGV (seems some kind of memory leak). 
(strace just lists the system calls by a process)

**Comment by diveshuttam** [comment]
Got the jekyll pages working. The log shows following
```
[2018-05-18 17:28:37] ERROR `/stylesheets/print.css' not found.
```
_print.css_ is linked in the following files as of now though it is not available in the stylesheets folder:
```
/egpg/_layouts/index.html
/egpg/_layouts/readme-2.0.html
/egpg/_layouts/readme-2.1.html
/egpg/_layouts/readme-2.2.html
```
Should we deal this as a separate issue or add to this only??
Also rest things seem to work properly but still have a look before merging.

**Comment by dashohoxha** [comment]
I merged it for the time being. We can fix any existing problems later.

**Pull request merged**